### ISSUE TITLE
_ensure_valid_dtype should also check for NETCDF4_CLASSIC

### ISSUE THREAD

**Issue by MartinDix** [SciTools/iris#2064]
In netcdf.py, `_ensure_valid_dtype` tries to convert types like int64 and unsigned that aren't supported in NetCDF3 to int32. I think the test should also include NETCDF4_CLASSIC.

E.g. saving a cube from a model start dump gives a file with an int64 variable (path here is to our collection of the data for UM rose stem testing)

```
cube = iris.load_cube('/g/data/access/TIDS/UM/standard_jobs/inputs/vn10.4/N48L70_EG/n48_eg_start_dump', 'air_potential_temperature')
iris.save(cube,'theta.nc')
```

theta.nc has

```
        int64 model_level_number(model_level_number) ;
```

Saving with

```
iris.save(cube,'theta.nc', netcdf_format='NETCDF3_64BIT')
```

gives 

```
       int model_level_number(model_level_number) ;
```

However trying to save as NETCDF4_CLASSIC

```
iris.save(cube,'theta.nc', netcdf_format='NETCDF4_CLASSIC')
```

fails with error

```
RuntimeError: NetCDF: Attempting netcdf-4 operation on strict nc3 netcdf-4 file
```

The netCDF documention http://www.unidata.ucar.edu/software/netcdf/docs/data_type.html says

> Only files created with the netCDF-4/HDF5 mode flag (NC_NETCDF4) but without the classic model flag (NC_CLASSIC_MODEL) may use user-defined types or the new atomic data types.

If I modify `_ensure_valid_dtype` to include NETCDF4_CLASSIC in the test things work as expected.

**Comment by ajdawson** [comment]
This sounds like a sensible approach to me. Would you like to submit a PR making this change @MartinDix?

**Comment by MartinDix** [comment]
#2080

**Comment by rcomer** [comment]
Looks like this was fixed by #2080.
